lib/onelogin/ruby-saml/utils.rb

```diff
@@ -86,6 +90,63 @@ def self.status_error_msg(error_msg, status_code = nil, status_message = nil)
 
         error_msg
       end
+
+      # Obtains the decrypted string from an Encrypted element
+      # @param encrypted_node [REXML::Element]     The Encrypted element
+      # @param private_key    [OpenSSL::PKey::RSA] The Service provider private key
+      # @return [String] The decrypted data
+      def self.decrypt_data(encrypted_node, private_key)
+        encrypt_data = REXML::XPath.first(encrypted_node, "./xenc:EncryptedData", { 'xenc' => XENC })
+        symmetric_key = retrieve_symmetric_key(encrypt_data, private_key)
+        cipher_value = REXML::XPath.first(encrypt_data, "//xenc:EncryptedData/xenc:CipherData/xenc:CipherValue", { 'xenc' => XENC })
+        node = Base64.decode64(cipher_value.text)
+        enc_method = REXML::XPath.first(encrypt_data, "//xenc:EncryptedData/xenc:EncryptionMethod", { 'xenc' => XENC })
+        algorithm = enc_method.attributes['Algorithm']
+        assertion_plaintext = retrieve_plaintext(node, symmetric_key, algorithm)        
+      end
+
+      # Obtains the symmetric key from the EncryptedData element
+      # @param encrypt_data [REXML::Element]     The EncryptedData element
+      # @param private_key [OpenSSL::PKey::RSA] The Service provider private key
+      # @return [String] The symmetric key
+      def self.retrieve_symmetric_key(encrypt_data, private_key)
+        encrypted_symmetric_key_element = REXML::XPath.first(encrypt_data, "//xenc:EncryptedData/ds:KeyInfo/xenc:EncryptedKey/xenc:CipherData/xenc:CipherValue", { "ds" => DSIG, "xenc" => XENC })
+        cipher_text = Base64.decode64(encrypted_symmetric_key_element.text)
+        enc_method = REXML::XPath.first(encrypt_data, "//xenc:EncryptedData/ds:KeyInfo/xenc:EncryptedKey/xenc:EncryptionMethod", {"ds" => DSIG,  "xenc" => XENC })
+        algorithm = enc_method.attributes['Algorithm']
+        retrieve_plaintext(cipher_text, private_key, algorithm)        
+      end
+
+      # Obtains the deciphered text
+      # @param cipher_text [String]   The ciphered text
+      # @param symmetric_key [String] The symetric key used to encrypt the text
+      # @param algorithm [String]     The encrypted algorithm
+      # @return [String] The deciphered text
+      def self.retrieve_plaintext(cipher_text, symmetric_key, algorithm)
+        case algorithm
```

Comment:
This method seems to belong to its own class. If you feel like it won't grow in time, then keep it here.